communicator/ssh/communicator.go

```diff
@@ -230,7 +230,7 @@ func (c *Communicator) UploadScript(path string, input io.Reader) error {
 	}
 
 	cmd := &remote.Cmd{
-		Command: fmt.Sprintf("chmod 0777 %s", c.connInfo.ScriptPath),
+		Command: fmt.Sprintf("chmod 0777 %s", path),
```

Comment:
Hm, I just glanced at the code again, `script` actually holds the content of the script. Woops, seems fine as is. (only reason our build is working is because shebang is parsed as comment and the error checking is just wrong after that.

An acceptance test would be great to have here to avoid brainfarts like mine.